account_analytic_parent/models/account_analytic_account.py

```diff
@@ -3,78 +3,125 @@
 # Copyright 2017 Matmoz d.o.o.
 # Copyright 2017 Deneroteam.
 # Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# Copyright 2018 Brainbean Apps
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
+from collections import defaultdict
+
 from odoo import _, api, fields, models
 from odoo.exceptions import UserError
 
 
 class AccountAnalyticAccount(models.Model):
-    _inherit = "account.analytic.account"
+    _inherit = 'account.analytic.account'
+    _parent_name = 'parent_id'
+    _parent_store = True
+    _rec_name = 'complete_name'
+    _order = 'complete_name'
 
+    parent_path = fields.Char(
+        index=True,
+    )
     parent_id = fields.Many2one(
-        comodel_name="account.analytic.account",
-        string="Parent Analytic Account"
+        string='Parent Analytic Account',
+        comodel_name='account.analytic.account',
+        index=True,
+        ondelete='cascade',
     )
     child_ids = fields.One2many(
-        comodel_name="account.analytic.account",
-        inverse_name="parent_id",
-        string="Child Accounts", copy=True)
+        string='Child Accounts',
+        comodel_name='account.analytic.account',
+        inverse_name='parent_id',
+        copy=True,
+    )
+    complete_name = fields.Char(
+        string='Complete Name',
+        compute='_compute_complete_name',
+        store=True,
+    )
 
     @api.multi
+    @api.depends('child_ids.line_ids.amount')
     def _compute_debit_credit_balance(self):
         """
         Warning, this method overwrites the standard because the hierarchy
         of analytic account changes
         """
-        super(AccountAnalyticAccount, self)._compute_debit_credit_balance()
-        analytic_line_obj = self.env['account.analytic.line']
-        # compute only analytic line
-        for account in self.filtered(lambda x: x.child_ids):
-            domain = [('account_id', 'child_of', account.ids)]
-            credit_groups = analytic_line_obj.read_group(
-                domain=domain + [('amount', '>', 0.0)],
-                fields=['account_id', 'amount'],
-                groupby=['account_id']
+        super()._compute_debit_credit_balance()
+
+        ResCurrency = self.env['res.currency']
+        AccountAnalyticLine = self.env['account.analytic.line']
+        user_currency_id = self.env.user.company_id.currency_id
+
+        # Re-compute only accounts with children
+        for account in self.filtered('child_ids'):
+            domain = [('account_id', 'child_of', account.id)]
+
+            credit_groups = AccountAnalyticLine.read_group(
+                domain=domain + [('amount', '>=', 0.0)],
+                fields=['currency_id', 'amount'],
+                groupby=['currency_id'],
+                lazy=False,
```

Comment:
I got an error on runbot when just clicking in the analytic account's menu to see the list view. It is in this line.

Reply:
This thing is resolved